packages/task/src/browser/task-service.ts

```diff
@@ -409,6 +410,46 @@ export class TaskService implements TaskConfigurationClient {
     }
 
     async runTask(task: TaskConfiguration, option?: RunTaskOption): Promise<TaskInfo | undefined> {
+        const runningTasksInfo: TaskInfo[] = await this.getRunningTasks();
+
+        // check if the task is active
+        const matchedRunningTaskInfo = runningTasksInfo.find(taskInfo => {
+            const taskConfig = taskInfo.config;
+            return this.taskDefinitionRegistry.compareTasks(taskConfig, task);
+        });
+        if (matchedRunningTaskInfo) { // the task is active
+            const taskName = this.taskNameResolver.resolve(task);
+            const taskId = matchedRunningTaskInfo.taskId;
+            const terminalId = matchedRunningTaskInfo.terminalId;
+            let terminal: TerminalWidget | undefined;
+            if (terminalId) {
+                terminal = this.terminalService.getById(this.getTerminalWidgetId(terminalId));
+                if (terminal) {
+                    terminal.show(); // bring the terminal to the top
+                }
+            }
+            this.messageService.info(`The task \'${taskName}\' is already active`, 'Terminate Task', 'Restart Task').then(async actionTask => {
+                if (actionTask) {
+                    await this.kill(taskId);
+                    if (terminal) {
+                        const parent = terminal.parent;
+                        terminal.close();
+                        if (actionTask === 'Terminate Task' && parent && parent.id === BOTTOM_AREA_ID) {
+                            parent.hide();
```

Comment:
The behavior can be improved for case when user has a few running tasks.
So if user terminates one task, but another task is running - output for second task should be displayed, we shouldn't hide the bottom area.
I don't mind to improve the behavior in another PR, up to you.